rust-src/concordium_base/src/transactions.rs

```diff
@@ -1464,6 +1465,52 @@ pub struct AccountAccessStructure {
     pub threshold: AccountThreshold,
 }
 
+type AccountStructure<'a> = &'a [(
+    CredentialIndex,
+    SignatureThreshold,
+    &'a [(KeyIndex, ed25519_dalek::PublicKey)],
+)];
+
+impl AccountAccessStructure {
+    /// Generate a new [`AccountAccessStructure`] for the thresholds, public
+    /// keys, and key indices specified in the input. If there are duplicate
+    /// indices then later ones override the previous ones.
+    pub fn generate(account_threshold: AccountThreshold, structure: AccountStructure) -> Self {
```

Comment:
I think it'd be better to call this `new` since we usually use `generate` for values that are somehow generated from either randomness or seed.